kin-ecosystem-core/src/main/java/kin/ecosystem/core/util/StringUtil.java

```diff
@@ -41,4 +42,9 @@ public static String toIndentedString(Object o) {
   public static String getAmountFormatted(int amount) {
     return NumberFormat.getNumberInstance(Locale.US).format(amount);
   }
+
+  public static String getAmountFormatted(String amount) {
+    int amountInt = new BigDecimal(amount).intValue();
+    return getAmountFormatted(amountInt);
```

Comment:
better try catch for NumberFormatException

Reply:
and return the original String - it will be visible in the UI but at least it will not crash